### ISSUE TITLE
Add an easier way to define a specific function name and similar in dialects

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#3088]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I've had to implement this in BigQuery:

```py
class ExtractFunctionNameSegment(BaseSegment):
    """EXTRACT function name segment.
    Need to be able to specify this as type function_name
    so that linting rules identify it properly
    """

    type = "function_name"
    match_grammar: Matchable = StringParser(
        "EXTRACT",
        CodeSegment,
        name="function_name_identifier",
        type="function_name_identifier",
    )
```

I don't want to use just `"EXTRACT"` and set it to an unreserved keyword, as I still want this to act as a `function_name_identifier` within `function_name` so tthe rules (L010, L017, and L030) still see the function name and can act upon that.

The code above works, but requires setting a new class and than a `StringParser` within that. Which seems a lot of code compared to using the `"EXTRACT"` keyword.

Is there an easier way? Should there be?

### Use case

Allow dialect grammer to be more easily written and understood for certain excepts (e.g. System Functions with certain structure)

### Dialect

All, but above example is from BigQuery.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
